### DISCUSSION TITLE
Device Recovery 8.3.1

### DISCUSSION THREAD

**Discussion by speedy-1702** [arendst/Tasmota#12460] [Q&A] [answered]
I have tasmotized a tuya device some months ago, now I need to reset the device following the doc, as described [here](https://tasmota.github.io/docs/Device-Recovery/)

Neither 40-seconds nor Fast Power Cycle works, other options are not feasible for me.

After trying the recovery procedure, the AP "tasmota-1670" is showing up, I can connect using my browser, but then credentials are requested in a UI, which I don't remember.

Assumption 1: When initially configuring tasmota I have created a password, otherwise credentials would not be requested?
Assumption 2: The device recovery methods described in the doc are valid also for tasmota 8.3.1?
Assumption 3: device recovery was not successful?

Any ideas or suggestions?

Attachment: UI
![Image 559](https://user-images.githubusercontent.com/1996354/123106565-00686c80-d439-11eb-96e9-df4912f3ccba.jpg)

**Comment by ascillato2** [comment, marked as answer]
If you don't remember the password you have set, you have to erase all flash and flash by serial.

**Comment by speedy-1702** [comment]
So, there is no recovery procedure available for tasmota 8.3.1.? 
Neither 40-seconds method nor Fast Power Cycle?

**Reply by ascillato** [reply]
Yes, any of those 2 methods reset everything to the default values. But if you have defined a password at compile time, that becomes the default. In that case, the only option is to erase all flash by serial and flash again.

**Reply by speedy-1702** [reply]
I didn't compile a binary on my own, instead I have flashed the tasmota-lite.bin, as recommended.
What else might interfere the 2 recovery procederes? 
I am pretty sure, that it didn't work properly.

**Reply by sfromis** [reply]
Both recovery procedures can be disabled depending on your setup. You could also try recovery again, sometimes people do not get the rhythm of the fast power cycle right, you should have each half-step no faster than 2 seconds.

**Reply by ascillato** [reply]
If you didn't compile and you did use the precompiled bin from this repository, they don't have passwords set. So, you should try the reset methods again.

The Fast Power Cycle is:

Disconnect your device and let it without power for 2 seconds. Then connect it again for 2 seconds and repeat this 2 steps 10 times. After that, your device should start in AP mode with ALL settings to firmware default.

**Reply by speedy-1702** [reply]
@sfromis, hmmm, I retried as suggested, without success. I am sure, not having disabled the recovery option in the configuration.

Could it be a good idea, retrying the power cycle directly with Vcc at the TYWE1S pcb rather than using the power button at the fan?

**Reply by speedy-1702** [reply]
@ascillato
**10** times, this is a new approach. Didn't tried this before.
Will do so as suggested and report back.

**Reply by sfromis** [reply]
"Using the power button at the fan" sounds wrong, depending on what exactly you mean by that. You should make sure to totally interrupt mains power input.

**Reply by ascillato** [reply]
The power is not the button. The recovery is the energy of the device. Your device has to be disconnected from the mains AC and must be reconnected 10 times waiting 2 seconds between actions. The power button is another things different than the required unplug and replug the device.

You need to unplug and replug the device

**Reply by speedy-1702** [reply]
_"Disconnect your device and let it without power for 2 seconds. Then connect it again for 2 seconds and repeat this 2 steps 10 times."_

Didn't help 😣

**Reply by speedy-1702** [reply]
_"You need to unplug and replug the device"_

Retried for the procedure, Didn't help 😣

**Reply by ascillato** [reply]
Ok, if that didn't trigger the full reset, means that while the device was online, intentionally or by accident you disable Tasmota's device recovery options using SetOption commands.

In that case, I'm sorry, you will need to reflash by wire.

**Reply by speedy-1702** [reply]
@ascillato @sfromis okay, will do so, thanks a lot so far. A CH340G is already ordered. After re-flashing and configuring wifi I will retry the recovery procedere and report back.

**Reply by speedy-1702** [reply]
Okay, did so as announced. Erased and re-flashed with Tasmotizer, configured wifi, then reset using 6 x off and on. And voila, recovery was done.

Thank you guys for your kind assistance.
